samples/subsys/usb/usb_pid.Kconfig

```diff
@@ -44,3 +44,7 @@ config USB_PID_WEBUSB_SAMPLE
 config USB_PID_BLE_HCI_SAMPLE
 	hex
 	default 0x000B
+
+config USB_PID_BLE_HCI_H4_SAMPLE
+	hex
+	default 0x000C
```

Comment:
Unrelated to this PR, but I had completely missed the existence of this kind of dependency from the USB subsystem to samples applications. Seems like it was added a year ago. Is this really the right way to go? It doesn't seem right to me that we would have sample app references from the core kernel or its subsystems. Why can't we just have per-sample "defaults" in each sample's prj.conf?

Samples are just that - samples. They act as a reference to the creation of final product applications, i.e. are mostly an ephemeral step in the app creation process. Having dedicated Kconfig options for them in the kernel or its subsystems feels therefore rather weird.

Reply:
I have similar thought, though we do need to track the PIDs in use for the samples otherwise it may confuse possible drivers if the samples end up reusing the same PID for multiple purposes, but like you said they are no suppose to be used in production, or at least we are not reserving this like that, anyway the application can overwrite this value if it want to use another PID, or perhaps we could have be using a single config option inform what PID shall be used then it is up to the sample itself to set it.

Reply:
I fully agree that we should coordinate (i.e. document) the PIDs that are used together with the Zephyr VID. That will help avoid conflicts when someone wants to do something custom with the Zephyr VID. I'd suggest we put this info somewhere in `doc/reference/usb`.